File: JRC_Abogados.Server/Models/Expediente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JRC_Abogados.Server.Models
{
    public class Expediente
    {
        [Key]
        public int Id { get; set; }
        public string Nombre { get; set; }
        public int TipoExpedienteId { get; set; }
        [ForeignKey("TipoExpedienteId")]
        public virtual TipoExpediente TipoExpediente { get; set; }
        public DateTime? FechaInicio { get; set; }
        public int ClienteId { get; set; }
        public int? CasoId { get; set; }
        [ForeignKey("CasoId")]
        public virtual Caso Caso { get; set; }
        [ForeignKey("ClienteId")]
        public virtual Cliente Cliente { get; set; }
        public int EmpleadoId { get; set; }
        [ForeignKey("EmpleadoId")]
        public virtual Empleado? Empleado { get; set; }
    }
}

File: jrc_abogados.client/src/app/Models/Expediente.ts
import { Caso } from "./Caso";
import { Cliente } from "./Cliente";
import { TipoExpediente } from "./TipoExpediente";

export class Expediente {
  id: number = 0;
  nombre: string = "";
  tipoExpedienteId = "";
  tipoExpediente: TipoExpediente = new TipoExpediente;
  fechaInicio = "";
  clienteId = "";
  casoId: any = '';
  caso: Caso = new Caso;
  cliente: Cliente = new Cliente;
  empleadoId: number = 0;
  empleado: any;
}
